### PR TITLE
Dao 4

### PR DESCRIPTION
**Pull request by entertainment-veks**

### CODE REVIEW

**Review by Fedich09 — APPROVED** [review]

File: src/main/java/dao3/service/impl/CarServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> result = new ArrayList<>();
+        for (Car currentCar : carDao.getAll()) {
+            for (Driver currentDriver : currentCar.getDrivers()) {
+                if (currentDriver.getId().equals(driverId)) {
+                    result.add(currentCar);
+                }
+            }
+        }
+        return result;
```
Comment (RomanDubovskyi) [inline src/main/java/dao3/service/impl/CarServiceImpl.java:63]: let's move the logic of getAllByDriver to the dao layer, we should receive filtered data from a dao

File: src/main/java/dao3/service/impl/CarServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        car.getDrivers().add(driver);
```
Comment (RomanDubovskyi) [inline src/main/java/dao3/service/impl/CarServiceImpl.java:45]:
```suggestion
        car.getDrivers().add(driver);
        carDao.update(car);
```
after we've modified a model we need to synchronize its' state with a db

File: src/main/java/dao3/service/impl/CarServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        car.getDrivers().remove(driver);
```
Comment (RomanDubovskyi) [inline src/main/java/dao3/service/impl/CarServiceImpl.java:50]: same here

**Review by RomanDubovskyi — APPROVED** [review]
fix my comments and then you can merge without a second review and start working on the next HW

**Pull request merged**